### PR TITLE
ArcCurve optimisation and fix

### PR DESCRIPTION
**Pull request by msdrey**
In ArcCurve:
- fixed a bug in getOffsetSegment: the control point was not computed correctly.
- fixed a typo in split()
- added a comment in getTangent to document the math
- rewrote computeCenter() in a simpler way by reusing the line intersection code

In Curve: 
- documented _computePoint()

### CODE REVIEW

File: src/canvas/Geometry/ArcCurve.tsx [hunk 1]
```diff
@@ -62,12 +62,17 @@ export class ArcCurve extends Curve {
         result.start = Point.translate(this.start, displacementOfStart);
         result.end = Point.translate(this.end, displacementOfEnd);
         result.radius = result.start.distanceTo(result.center);
-        result.control = Point.translate(Point.translate(this.control, displacementOfStart), displacementOfEnd); 
+
+        const lengthOfDisplacementOfControl = distance / Math.cos(Math.abs(this.endAngle - this.startAngle) / 2);
```
Comment (SarahPappas) [inline src/canvas/Geometry/ArcCurve.tsx:66]: Why did you add these lines?

Comment (msdrey) [reply src/canvas/Geometry/ArcCurve.tsx:66]: The old way of computing the new control point was wrong. That was something we noticed back in september, and I thought I had fixed it, but apparently it didn't make it to the master branch. It was causing a bug with smaller sections of offset arc curves, they were wonky. The new lines are the new math, which I now think is correct.

Comment (SarahPappas) [reply src/canvas/Geometry/ArcCurve.tsx:66]: Ok, thanks!
